### PR TITLE
http-api: T5126: allow restricting client IP address

### PR DESCRIPTION
**Pull request by jestabro**
## Change Summary

Add multi-value 'service https virtual-host example allow-client address <ipv4|ipv6|ipv4net|ipv6net>' to restrict client connections to only those specified.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://vyos.dev/T5126

## Checklist:

- [X] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [X] I have linked this PR to one or more Phabricator Task(s)
- [X] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [X] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/https.xml.in [hunk 1]
```diff
@@ -60,6 +60,32 @@
                   <multi/>
                 </properties>
               </leafNode>
+              <leafNode name="allow-client">
```
Comment (sever-sever) [inline interface-definitions/https.xml.in]: I wonder if there should be `node` before, I.e. `allow-client address x.x.x.x` vs `allow-client x.x.x.x`

as we already have `allow-client address x.x.x.x` for other commands we can add XML and reuse them for new XML via “include”

@dmbaturin @c-po your thoughts?

Comment (jestabro) [reply interface-definitions/https.xml.in]: Good point, @sever-sever; I'll fix.

Comment (c-po) [reply interface-definitions/https.xml.in]: Good idea @sever-sever  +1

Comment (jestabro) [reply interface-definitions/https.xml.in]: Fixed; thanks @sever-sever

**Review by sever-sever — APPROVED** [review]

File: interface-definitions/https.xml.in [hunk 2]
```diff
@@ -60,6 +60,39 @@
                   <multi/>
                 </properties>
               </leafNode>
+              <node name="allow-client">
```
Comment (sever-sever) [inline interface-definitions/https.xml.in]: Can we use it as a separate XML file and include it. So we can re-use this for other services
```
#include <include/allow-client.xml.i>
```

Comment (jestabro) [reply interface-definitions/https.xml.in:63]: But I'm happy to make it an include file, if you prefer.

Comment (jestabro) [reply interface-definitions/https.xml.in]: Sure, I'll move it to include.

Comment (jestabro) [reply interface-definitions/https.xml.in]: Done; rebased over current.

Comment (sever-sever) [reply interface-definitions/https.xml.in]: Thanks!

**Pull request merged**